hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/semaphore/client/GetSemaphoreTypeMessageTask.java

```diff
@@ -59,7 +60,7 @@ public String getServiceName() {
 
     @Override
     public Permission getRequiredPermission() {
-        return null;
+        return new SemaphorePermission(parameters.proxyName);
```

Comment:
Shouldn't this be parameters.name?

Reply:
`parameters.name` is not an available attribute in this context, and `parameters.proxyName` aligns with usage in other methods like `getDistributedObjectName()`.  so I believe using `parameters.proxyName` is the correct approach in this case.